src/plugins/home/server/services/sample_data/routes/install.ts

```diff
@@ -217,6 +218,9 @@ export function createInstallRoute(
       } catch (err) {
         const errMsg = `bulkCreate failed, error: ${err.message}`;
         logger.warn(errMsg);
+        if (workspaceId && SavedObjectsErrorHelpers.isForbiddenError(err)) {
```

Comment:
do we have any other error except forbidden error?

Reply:
I've searched the whole org, there are [three places](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards%20SavedObjectsErrorHelpers.decorateForbiddenError&type=code) will generate ForbiddenError:
1. OpenSearch response forbidden error: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/32b11de6e20f3d233a0c337a7ab0a36588d54c72/src/core/server/saved_objects/service/lib/decorate_opensearch_error.ts#L80
2. Workspace saved object permission wrapper forbidden error: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/32b11de6e20f3d233a0c337a7ab0a36588d54c72/src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts#L47
3. Data source permission wrapper forbidden error: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/32b11de6e20f3d233a0c337a7ab0a36588d54c72/src/plugins/data_source_management/server/saved_objects/data_source_premission_client_wrapper.ts#L163
I think it's OK to response 403 for case 2 and 3, seems they are user permission related issues. 
For case 1, the server response 500 in the old implementation. The current implementation will return 403. 
We may need to check the error message if we want to distinguish these cases. Do you have other suggestions? I think response 403 would be OK for case 1.

Reply:
Nit: Why do we check if there is workspaceId present? I think returning 403 error if the error is a forbidden error seems straightforward to me.